### PR TITLE
Clear button in searchInput 

### LINKED ISSUE #236: When filtering returns no results show that to the user
image

Add `EmptyState` which shows `No files or directories found`.

As a bonus add a cross button which removes the current filter as in the hosts selector:

image

### PR DESCRIPTION
**Pull request by Abdelra6**
closes #236 

replicate of #240 closed it by accident because of some conflicts

### CODE REVIEW

**Comment by Abdelra6** [comment]
@garrett the width of the input is fixed now. I changed the components. I hope that's a better one)

**Comment by garrett** [comment]
Oh, we usually re-use the same PR for the same branch so the conversation isn't fragmented and gets lost. That's done with a force push, such as `git push --force-with-lease` and the branch info (this version checks to make sure your local branch is up to date with the server).

Two problems:

1. The X icon is always there now, even when nothing was typed.
2. When clicking the X to clear the entry, I get an "Ooops!" error.

image

My dev console says: 

```
Uncaught TypeError: setCurrentFilter is not a function
    onClick header.jsx:62
    React 15
        callCallback2
        invokeGuardedCallbackDev
        invokeGuardedCallback
        invokeGuardedCallbackAndCatchFirstError
        executeDispatch
        processDispatchQueueItemsInOrder
        processDispatchQueue
        dispatchEventsForPlugins
        dispatchEventForPluginEventSystem
        batchedUpdates$1
        batchedUpdates
        dispatchEventForPluginEventSystem
        dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay
        dispatchEvent
        dispatchDiscreteEvent
```

**Review by garrett — CHANGES_REQUESTED** [review]
(See above; I forgot to make it a review instead of a normal comment.)

**Comment by Abdelra6** [comment]
Sorry I can't test it completely due to the fsinfo commit. How can I make the [x] button appear without changing the width of the input?

File: src/header.jsx [hunk 1]
```diff
@@ -49,10 +52,20 @@ export const NavigatorCardHeader = ({ currentFilter, onFilterChange, isGrid, set
...
-                  onChange={onFilterChange}
-                />
+                <TextInputGroup>
+                    <TextInputGroupMain
+                      icon={<SearchIcon />} value={currentFilter}
+                      onChange={onFilterChange} placeholder={_("Filter directory")}
+                    />
+                    <TextInputGroupUtilities>
+                        <Button
+                          variant="plain" onClick={_ => setCurrentFilter("")}
```
Comment (jelly) [inline src/header.jsx:62]: `_` is already bound to `cockpit.gettext`. Just pass setCurrentFilter("").

File: src/header.jsx [hunk 2]
```diff
@@ -49,10 +52,20 @@ export const NavigatorCardHeader = ({ currentFilter, onFilterChange, isGrid, set
...
-                <SearchInput
-                  placeholder={_("Filter directory")} value={currentFilter}
-                  onChange={onFilterChange}
-                />
+                <TextInputGroup>
+                    <TextInputGroupMain
+                      icon={<SearchIcon />} value={currentFilter}
+                      onChange={onFilterChange} placeholder={_("Filter directory")}
+                    />
+                    <TextInputGroupUtilities>
```
Comment (jelly) [inline src/header.jsx:60]: This needs to be behind some conditional like `currentFilter !== ""`

**Comment by jelly** [comment]
Superseeded by https://github.com/cockpit-project/cockpit-files/pull/465

**Pull request closed without merging**